presto-hive/src/main/java/io/prestosql/plugin/hive/rubix/PrestoClusterManager.java

```diff
@@ -0,0 +1,78 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.prestosql.plugin.hive.rubix;
+
+import com.google.common.collect.ImmutableList;
+import com.qubole.rubix.spi.ClusterManager;
+import com.qubole.rubix.spi.ClusterType;
+import io.prestosql.spi.Node;
+import io.prestosql.spi.NodeManager;
+
+import java.util.List;
+import java.util.Set;
+
+import static com.google.common.collect.ImmutableList.toImmutableList;
+import static com.qubole.rubix.spi.ClusterType.PRESTOSQL_CLUSTER_MANAGER;
+import static java.util.Objects.requireNonNull;
+
+public class PrestoClusterManager
+        extends ClusterManager
+{
+    private static NodeManager nodeManager;
+
+    static void setNodeManager(NodeManager nodeManager)
+    {
+        PrestoClusterManager.nodeManager = requireNonNull(nodeManager, "nodeManager is null");
+    }
+
+    @Override
+    public ClusterType getClusterType()
+    {
+        return PRESTOSQL_CLUSTER_MANAGER;
+    }
+
+    @Override
+    public boolean isMaster()
+    {
+        return nodeManager.getCurrentNode().isCoordinator();
+    }
+
+    @Override
+    public List<String> getNodes()
+    {
+        Set<Node> workers = nodeManager.getWorkerNodes();
+        if (workers.isEmpty()) {
+            // Empty result set => server up and only master node running, return localhost has the only node
+            // Do not need to consider failed nodes list as 1node cluster and server is up since it replied to allNodesRequest
+            return ImmutableList.of(nodeManager.getCurrentNode().getHost());
+        }
+
+        return nodeManager.getWorkerNodes().stream()
```

Comment:
How frequently is that called? I guess constructing list on each call should not be a problem but please make sure it is not.